### PR TITLE
predict() support for segmentation models and more

### PR DESCRIPTION
**Pull request by BloodAxe**
Here we go again with predict() support for segmentation models.
What have I done on top of former PR:

* Deprecated SegToTensor in favor of SegConvertToTensor. Old class disallow it's creation and raise an error to leave no options other than migrate. The SegToTensor was doing /255 and HWC->CHW which could cause many unwanted issues since /255 was performed only when input is byte dtype. There are many cases when image dtype can change unintentionally. Moreover, SegToTensor had to be applied first and only after SegNormalize. And it was breaking preprocessing in predict().
* Introduce SegStandardize
* Add missing default preprocessing params for other segmentation models
* Improves visualization of segmentation mask when the task is binary segmentation
* Implement missing methods across the segmentation transforms 
* All recipes updated to use SegConvertToTensor
* SegmentationSample will convert PIL image under the hood to np.ndarray and Segmentaiton transforms will not return PIL image anymore

### CODE REVIEW

**Review by ofrimasad — COMMENTED** [review]
Very nice!
a few minor comments.
Does this support long video (like the new implementation in object detection)?

File: src/super_gradients/training/models/segmentation_models/stdc.py [hunk 1]
```diff
@@ -636,6 +643,53 @@ def replace_input_channels(self, in_channels: int, compute_new_weights_fn: Optio
...
+    ) -> None:
+        """Set the processing parameters for the dataset.
+
+        :param class_names:     (Optional) Names of the dataset the model was trained on.
+        :param image_processor: (Optional) Image processing objects to reproduce the dataset preprocessing used for training.
+        """
+        self._class_names = class_names or self._class_names
+        self._image_processor = image_processor or self._image_processor
+
+    def _get_pipeline(self, fuse_model: bool = True) -> SegmentationPipeline:
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/training/models/segmentation_models/stdc.py:662]: Fixed in 57851550ab9fa46368eba46186774496ea741895

File: src/super_gradients/examples/predict/segmentation_predict.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from super_gradients.common.object_names import Models
+from super_gradients.training import models
+
+
+model = models.get(model_name=Models.PP_LITE_T_SEG75, arch_params={"use_aux_heads": False}, num_classes=19, pretrained_weights="cityscapes")
+
+IMAGES = [
+    "../../../../documentation/source/images/examples/countryside.jpg",
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/examples/predict/segmentation_predict.py]: Fixed in 99589492f53edbebe4911f3c0105ab668fc5855a

File: src/super_gradients/common/object_names.py [hunk 3]
```diff
@@ -441,3 +443,8 @@ class Processings:
     ComposeProcessing = "ComposeProcessing"
     CenterCrop = "CenterCrop"
     Resize = "Resize"
+    SegResizeWithPadding = "SegResizeWithPadding"
+    SegmentationRescale = "SegmentationRescale"
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/common/object_names.py:448]: Fixed in 3c6b4706e3b31aeeee5f7842805991487af3c514
The segmentation transforms has Seg prefix to keep the compatibility (no change here)
Processings has "Segmentation" prefix to be consistent with other processings (Detection and Keypoint) that has full name prefix

**Comment by BloodAxe** [comment]
> Very nice! a few minor comments. Does this support long video (like the new implementation in object detection)?

Yes, the way video prediction is implemented is actually pretty nice since all the tasks support long video processing automatically.

File: src/super_gradients/examples/predict/segmentation_predict.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from super_gradients.common.object_names import Models
+from super_gradients.training import models
+
+
+model = models.get(model_name=Models.PP_LITE_T_SEG75, pretrained_weights="cityscapes")
+
+IMAGES = [
+    "https://deci-pretrained-models.s3.amazonaws.com/sample_images/beatles-abbeyroad.jpg",
+]
```
Comment (Louis-Dupont) [inline src/super_gradients/examples/predict/segmentation_predict.py:9]: Not blocking, but it's nicer if the url would map to github (we updated notebooks to work like that).
You can find the images here https://github.com/Deci-AI/super-gradients/tree/master/documentation/source/images/examples

File: src/super_gradients/training/processing/processing.py [hunk 5]
```diff
@@ -668,6 +684,202 @@ def resizes_image(self) -> bool:
...
+
+
+@register_processing(Processings.SegmentationPadShortToCropSize)
+class SegmentationPadShortToCropSize(Processing):
+    """
+        Pads image to 'crop_size'.
+        Should be called only after "SegRescale" or "SegRandomRescale" in augmentations pipeline.
+
+        :param crop_size:   Tuple of (width, height) for the final crop size, if is scalar size is a square (crop_size, crop_size)
+    =    :param fill_image:  Grey value to fill image padded background.
```
Comment (Louis-Dupont) [inline src/super_gradients/training/processing/processing.py:815]: Too much indented and include a typo (`=`)

File: src/super_gradients/training/processing/processing.py [hunk 6]
```diff
@@ -668,6 +684,202 @@ def resizes_image(self) -> bool:
...
+        return True
+
+
+@register_processing(Processings.SegmentationPadToDivisible)
+class SegmentationPadToDivisible(Processing):
+    """
+        Pads image to a size divisible by the defined parameter.
+
+        :param divisible_value:   the divisible value, new image size is an int multiplication of this number
+    =    :param fill_image:  Grey value to fill image padded background.
```
Comment (Louis-Dupont) [inline src/super_gradients/training/processing/processing.py:851]: Too much indented and include a typo (`=`)

**Review by Louis-Dupont — COMMENTED** [review]
Left a few small comments

File: src/super_gradients/training/processing/processing.py (hunk 6 above)
Comment (BloodAxe) [reply src/super_gradients/training/processing/processing.py]: Thanks for spotting! Fixed in 24538f292d59682d4a54ab82db048b62c62dddeb

File: src/super_gradients/training/pipelines/pipelines.py [hunk 7]
```diff
@@ -488,3 +492,74 @@ def _combine_image_prediction_to_video(
...
+                    image_shape=image.shape[-2:],
+                )
+            )
+
+        return predictions
+
+    def _instantiate_image_prediction(self, image: np.ndarray, prediction: DetectionPrediction) -> ImagePrediction:
+        return ImageSegmentationPrediction(image=image, prediction=prediction, class_names=self.class_names)
+
+    def _combine_image_prediction_to_images(
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/training/pipelines/pipelines.py:556]: Can be, but i think I tend to keep it that way to be consistent with the other predict tasks implementation where this method is non-static

**Review by Louis-Dupont — APPROVED** [review]
LGTM

**Pull request merged**